sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/BinaryDataRequestBody.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.http.okhttp.implementation;
+
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.logging.ClientLogger;
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+import okio.BufferedSink;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+/**
+ * Implementation of {@link RequestBody} where the content is a {@link BinaryData}.
+ */
+public class BinaryDataRequestBody extends RequestBody {
+    private static final ClientLogger LOGGER = new ClientLogger(BinaryDataRequestBody.class);
+
+    private final MediaType contentType;
+    private final BinaryData body;
+    private final long effectiveContentLength;
+
+    private volatile int bodySent = 0;
```

Comment:
Can we use `AtomicBoolean` instead?

Reply:
I went with this as it reduces the size of the object and is effectively the same code AtomicBoolean uses